Source/WebKit/Configurations/WebKit.xcconfig

```diff
@@ -222,6 +222,9 @@ OTHER_LDFLAGS = $(inherited) -lsqlite3 -iframework $(SDK_DIR)$(SYSTEM_LIBRARY_DI
 
 OTHER_LDFLAGS_DELAY_INIT[sdk=iphone*] = -Wl,-delay_framework,CoreTelephony;
 OTHER_LDFLAGS_DELAY_INIT[sdk=iphone*17.*] = ;
+OTHER_LDFLAGS_DELAY_INIT[sdk=appletv*] = ;
+OTHER_LDFLAGS_DELAY_INIT[sdk=watch*] = ;
+OTHER_LDFLAGS_DELAY_INIT[sdk=xr*] = ;
```

Comment:
The problem here is that these platforms fall back to the `iphoneos` sdk conditional, right? Maybe it would be better to use WK_PLATFORM_NAME, i.e.
```
OTHER_LDFLAGS_DELAY_INIT = $(OTHER_LDFLAGS_DELAY_INIT_$(WK_PLATFORM_NAME));
OTHER_LDFLAGS_DELAY_INIT_iphoneos = -Wl,-delay_framework,CoreTelephony;
OTHER_LDFLAGS_DELAY_INIT_iphonesimulator = -Wl,-delay_framework,CoreTelephony;
```
to avoid the fallthrough.

Reply:
> The problem here is that these platforms fall back to the `iphoneos` sdk conditional, right?

That is correct.

I think using `WK_PLATFORM_NAME` would work, but because delay-init linking is only available in the mysterious future, we still need to turn it off for older OS versions, so we’d need something like this:

```
OTHER_LDFLAGS_DELAY_INIT = $(OTHER_LDFLAGS_DELAY_INIT_$(WK_PLATFORM_NAME));
OTHER_LDFLAGS_DELAY_INIT_iphoneos = -Wl,-delay_framework,CoreTelephony;
OTHER_LDFLAGS_DELAY_INIT_iphoneos[sdk=iphoneos17.*] = ;
OTHER_LDFLAGS_DELAY_INIT_iphonesimulator = -Wl,-delay_framework,CoreTelephony;
OTHER_LDFLAGS_DELAY_INIT_iphonesimulator[sdk=iphonesimulator17.*] = ;
```

Ultimately, if we add another platform, I think this will end up being more verbose because we can’t combine OS and Simulator together.

However, if we need a delay-init link on only one of either OS or Simulator, then `WK_PLATFORM_NAME` would help.

Let’s keep this in mind for the future—we can always change it later.